### ISSUE TITLE
Closing editor programmatically locks grid

### ISSUE THREAD

**Issue by onlineandy** [TatuLund/GridFastNavigation#88]
When I call ``grid.getEditor().cancel()`` while the editor ist open, the editor gets closed. But afterwards the grid is locked. In HTML I can see that the "curtain" is not removed.

**Comment by TatuLund** [comment]
Which version of the add-on you are using? I remember that I fixed one issue regarding modality curtain not removed on close in version 2.2.1, see: https://github.com/TatuLund/GridFastNavigation/issues/67

**Comment by onlineandy** [comment]
I am using 2.3.10.

**Comment by TatuLund** [comment]
Can you give an example how you replicated this issue. I tried to do it myself, but could not replicate. What kind of symptoms you see?

**Comment by onlineandy** [comment]
Basically I am trying to close the editor from another button in another row. 

I could replicate it in your demo app when adding the following column:

```
this.addComponentColumn(new ValueProvider<DemoColumns, Button>() {
			@Override
			public Button apply(DemoColumns arg0) {
				Button b = new Button("Click");
				b.addClickListener(event -> DemoFastGrid.this.getEditor().cancel());
				return b;
			}
		});
```

**Comment by TatuLund** [comment]
Ok, that helped me to reproduce the issue. I checked how cancel actually works, and it is implemented in the way that it does not produce call to EditorStateManager's handleCloseEvent. Hence the code that removes the modality curtain is newer called. 

What could be a workaround is to introduce cancel() method in the extension as well, which does normal cancel procedure and takes care of removing the modality curtain as well.

**Comment by TatuLund** [comment]
I did some further study on this. And I found that it is not that easy. If I close editor via extension in the click event of the rendered button, something is left still in messed state in Editor and I think it is in Grid itself in the way I can't handle via extension. When I click to re-open the editor, it does not fully open (it tries to, but goes half way and modality curtain is back). When the closing is not done via rendered button, things are normal both with regular Editor#cancel() method or closing via extension. So I think this is a corner case regarding just closing editor via rendered button.
